validator/remote/src/main/java/tech/pegasys/teku/validator/remote/typedef/handlers/ProduceBlockRequest.java

```diff
@@ -91,7 +94,29 @@ public ProduceBlockRequest(
 
     this.responseHandler =
         new ResponseHandler<>(produceBlockTypeDefinition)
-            .withHandler(SC_OK, this::handleBlockContainerResult);
+            .withHandler(SC_OK, this::handleBlockContainerResult)
+            .withHandler(SC_NOT_FOUND, this::fallbackToBlockV2);
+  }
+
+  private Optional<ProduceBlockResponse> fallbackToBlockV2(Request request, Response response) {
+    LOG.warn("Block V3 request failed. Retrying with Block V2");
```

Comment:
I think we'd better do something like this https://github.com/mehdi-aouadi/teku/blob/30d80074165142e9ca7035cff27462d283708e4a/validator/remote/src/main/java/tech/pegasys/teku/validator/remote/typedef/OkHttpValidatorTypeDefClient.java#L100-L108
So, create an exception, something like `NewBlockV3NotSupportedException` (not a best name), in NOT_FOUND handler log warning and throw it, in `OkHttpValidatorTypeDefClient` catch it and go V2

Reply:
Yep definitely way cleaner that way. Done

Reply:
The only thing to notice is that we always fallback to the blinded block production since we don't know if block v3 was going to produce a blinded or an unblinded block... But anyway this fallback is temporary and should go once we switch to block v3